src/components/activity/ActivityRegistrator.tsx

```diff
@@ -10,36 +10,101 @@ import { ChoiceBasedTextInput } from '../ChoiceBasedTextInput';
 
 import { useTranslation } from 'language/LanguageProvider';
 import Storage from 'storage';
-import { ActivitiesEntry } from 'storage/types';
+import { ActivitiesDay, ActivitiesEntry } from 'storage/types';
 
 import { ISODate } from 'utils';
+import { getDaysInMonth } from 'react-native-paper-dates/lib/typescript/src/Date/dateUtils';
 
 export const ActivityRegistrator = ({ route, navigation }: any) => {
   const lang = useTranslation();
   const [settings, modifySettings] = Storage.useSettings();
   const { iconSizes, colors } = useTheme();
+  const [values, modifyValues] = Storage.useValues();
+  const [activities, modifyActivities] = Storage.useActivities();
 
   const steps = 1;
+  let dateDefault = new Date();
+  let fromTimeRoundedDefault = getCurrentTimeRounded(0, steps);
+  let toTimeRoundedDefault = getCurrentTimeRounded(1, steps);
+  let activityTextDefault = '';
+  let importanceDefault = 5;
+  let enjoymentDefault = 5;
 
-  const [values, modifyValues] = Storage.useValues();
-  const [activities, modifyActivities] = Storage.useActivities();
+  // Overwrite default values from entry (if it exists)
+  if (route.params?.entry && route.params?.day) {
+    const day: ActivitiesDay = route.params?.day;
+    const entry: ActivitiesEntry | null = route.params?.entry;
+    let startEntry: ActivitiesEntry | null = null;
+    let endEntry: ActivitiesEntry | null = null;
+    let startIndex = 0;
+    let endIndex = 0;
+
+    for(let i = 0; i < day.entries.length; ++i) {
+      const currentEntry = day.entries[i];
+      // If element matches entry
+      if(currentEntry === entry) {
```

Comment:
Entries cannot be compared with the equality operator.

Reply:
This code has been entirely reworked.